factory/builder.py

```diff
@@ -225,9 +225,16 @@ def chain(self):
         return (self.stub,) + parent_chain
 
     def recurse(self, factory, declarations, force_sequence=None):
+        if not hasattr(factory, '_meta') or not hasattr(factory._meta, 'factory'):
+            raise errors.AssociatedClassError(
+                "%r: Attempting to recursing into a non-factory object %r"
+                % (self, factory))
```

Comment:
how about `issubclass(factory, Factory)`?

Reply:
That's a good point; it brings a localized import, but the gains in readability are worth it.